base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/views/ScriptGeneratorViewModel.java

```diff
@@ -1033,4 +1040,42 @@ public boolean getHasSelection() {
 	public void setRepoPath() {
 		scriptGeneratorModel.setRepoPath();
 	}
+	
+	/**
+	 * Copies actions to clipboard.
+	 * @param actions copied actions
+	 */
+	public void copyActions(String actions) {
+		clipboard.setContents(new Object[] {actions}, new Transfer[] {TextTransfer.getInstance()});
+	}
+	
+	/**
+	 * Paste copied actions to the script generator. If the user wants to paste the copied actions
+	 * to another script definition, it will decide how many values(per row) to paste depending on the number
+	 * column label in new script definition. if there are 10 copied values and the script definition to
+	 * which user has switched to only contains 3 param, it will paste only first 3 values however,
+	 * all 10 copied values will still be in clipboard.
+	 * @param columnsLabel list of column labels present in current table
+	 * @param pasteLocation row where user wants to paste.
+	 */
+	public void pasteActions(ArrayList<String> columnsLabel, int pasteLocation) {
+		String copiedActions = (String) clipboard.getContents(TextTransfer.getInstance());
+		// remove constant labels
+		columnsLabel.removeAll(Arrays.asList("Line", "Validity", "Estimated run time"));
```

Comment:
It's not good to have the column names here. It means if we have to change a column name for some reason or add a new un-editable column we would have to do it in two places. Better to use the `NON_EDITABLE_COLUMNS` variables I think.